### PR TITLE
Added separate add for all reference types

### PR DESCRIPTION
**Pull request by saurabsa**

### CODE REVIEW

File: src/command_modules/azure-cli-project/azure/cli/command_modules/project/_params.py [hunk 1]
```diff
@@ -51,25 +51,25 @@
...
-    help="Space separated 'name=value' pairs",
-    required=False)
+    help="Space separated 'name=value' pairs"
+)
 
 optional_target_group = CliArgumentType(
     options_list=('--target-group', '-g'),
-    help="Target resoure group name",
-    required=False
+    help="Target resoure group name"
```
Comment (YuzorMa) [inline src/command_modules/azure-cli-project/azure/cli/command_modules/project/_params.py]: typo: resource not resoure

File: src/command_modules/azure-cli-project/azure/cli/command_modules/project/commands.py [hunk 2]
```diff
@@ -11,9 +11,15 @@
...
-cli_command(__name__, 'project reference add',
-            'azure.cli.command_modules.project.custom#add_reference')
-cli_command(__name__, 'project reference remove',
+cli_command(__name__, 'project add existing mongo',
+            'azure.cli.command_modules.project.custom#add_reference_mongo')
+cli_command(__name__, 'project add existing sql',
+            'azure.cli.command_modules.project.custom#add_reference_sql')
+cli_command(__name__, 'project add existing servicebus',
+            'azure.cli.command_modules.project.custom#add_reference_servicebus')
+cli_command(__name__, 'project add existing generic',
```
Comment (YuzorMa) [inline src/command_modules/azure-cli-project/azure/cli/command_modules/project/commands.py]: I don't know that I like 'generic'.  We should check with John to see what he wants

Comment (saurabsa) [inline src/command_modules/azure-cli-project/azure/cli/command_modules/project/commands.py]: Till now we had this generic type when target name and group is not provided. We need to decide that do we still need it?

Comment (YuzorMa) [inline src/command_modules/azure-cli-project/azure/cli/command_modules/project/commands.py]: The functionality makes sense, I just don't think we would call it 'generic' for the user.  Need to get John's input on what the word should be.

**Pull request merged**